sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -1773,9 +1778,14 @@ void gcoap_forward_proxy_find_req_memo(gcoap_request_memo_t **memo_ptr,
     _find_req_memo(memo_ptr, src_pdu, remote, false);
 }
 
-ssize_t gcoap_forward_proxy_dispatch(const uint8_t *buf, size_t len, sock_udp_ep_t *remote)
+void gcoap_forward_proxy_post_event(void *arg)
+{
+    event_post(&_queue, arg);
+}
+
+uint16_t gcoap_forward_proxy_get_msg_id(void)
 {
-    return sock_udp_send(&_sock_udp, buf, len, remote);
+    return (uint16_t)atomic_fetch_add(&_coap_state.next_message_id, 1);
```

Comment:
This doesn't really have anything to do with the forward proxy, right?
Might also be used by some other GCoAP module in the future

Reply:
Currently, it is exposed to `gcoap_forward_proxy` via extern. As this is for gcoap internal use only, I think it is better to keep it not exposed in the header.

Reply:
Yea no need to put it in the header.